xpart/matched_gaussian.py

```diff
@@ -111,18 +111,127 @@ def generate_matched_gaussian_bunch(num_particles,
         # go to particles.weight = 1
         total_intensity_particles = num_particles
 
-
     part = build_particles(_context=_context, _buffer=_buffer, _offset=_offset,
-                      R_matrix=R_matrix,
-                      particles_class=particles_class,
-                      particle_on_co=particle_on_co,
-                      particle_ref=(
-                          particle_ref if particle_on_co is  None else None),
-                      line=line,
-                      zeta=zeta, delta=delta,
-                      x_norm=x_norm, px_norm=px_norm,
-                      y_norm=y_norm, py_norm=py_norm,
-                      nemitt_x=nemitt_x, nemitt_y=nemitt_y,
-                      weight=total_intensity_particles/num_particles,
-                      **kwargs)
+                           R_matrix=R_matrix,
+                           particles_class=particles_class,
+                           particle_on_co=particle_on_co,
+                           particle_ref=(
+                               particle_ref if particle_on_co is None
+                               else None),
+                           line=line,
+                           zeta=zeta, delta=delta,
+                           x_norm=x_norm, px_norm=px_norm,
+                           y_norm=y_norm, py_norm=py_norm,
+                           nemitt_x=nemitt_x, nemitt_y=nemitt_y,
+                           weight=total_intensity_particles / num_particles,
+                           **kwargs)
     return part
+
+
+def split_scheme(filling_scheme, communicator=None):
+    """
+    Distribute the filling scheme between the processes, i.e. assign to each
+    processor its bunches
```

Comment:
Use proper "numpy" style docstring with description of method arguments and outputs